metabase/metabase
# feat(sdk): Add Dashboard creation components

Closes [Embedding SDK Dashboard creation](https://www.notion.so/metabase/Expose-front-end-components-in-SDK-e2f96fa53cf043bda66002898bedf10c?pvs=4#322ebed58a10424b9f7ef1309601a7fe)

### Description

This adds `DashboardCreateModal` component and `useDashboardCreate` hook to kick-off dashboard creation, which then can be edited with `EditableDashboard` component added in https://github.com/metabase/metabase/pull/46255

### How to verify

1. Run Embedding SDK storybook - `yarn storybook-embedding-sdk`
2. Check DashboardCreateModal stories - `http://localhost:6006/?path=/story/embeddingsdk-dashboardcreatemodal--default`

Then also can checked in Shoppy:
1. Build the sdk - `yarn build-embedding-sdk`
2. Checkout https://github.com/metabase/shoppy/tree/testing-dashboard
3. Run `yarn install-and-start`
4. Open `https://localhost:3004/dash`

### Demo


https://github.com/user-attachments/assets/09f2c57f-ef0a-44ac-990e-02a287fb0e7d



### Checklist
